core/session.ts

```diff
@@ -335,34 +346,32 @@ export class Session {
   public compile(): dataform.ICompiledGraph {
     const compiledGraph = dataform.CompiledGraph.create({
       projectConfig: this.config,
-      tables: this.compileGraphChunk(this.tables),
-      operations: this.compileGraphChunk(this.operations),
-      assertions: this.compileGraphChunk(this.assertions),
-      tests: this.compileGraphChunk(this.tests),
+      tables: this.compileGraphChunk(this.actions.filter(action => action instanceof table.Table)),
+      operations: this.compileGraphChunk(
+        this.actions.filter(action => action instanceof Operation)
+      ),
+      assertions: this.compileGraphChunk(
+        this.actions.filter(action => action instanceof Assertion)
+      ),
+      tests: this.compileGraphChunk(Object.values(this.tests)),
       graphErrors: this.graphErrors
     });
 
-    // Expand action dependency wildcards.
-
     const allActions: IActionProto[] = [].concat(
       compiledGraph.tables,
       compiledGraph.assertions,
       compiledGraph.operations
     );
-    const allActionNames = allActions.map(action => action.name);
 
     allActions.forEach(action => {
-      const uniqueDependencies: { [dependency: string]: boolean } = {};
-      const dependencies = action.dependencies || [];
-      // Add non-wildcard deps normally.
-      dependencies
-        .filter(dependency => !dependency.includes("*"))
-        .forEach(dependency => (uniqueDependencies[dependency] = true));
-      // Match wildcard deps against all action names.
-      utils
-        .matchPatterns(dependencies.filter(d => d.includes("*")), allActionNames)
-        .forEach(dependency => (uniqueDependencies[dependency] = true));
-      action.dependencies = Object.keys(uniqueDependencies);
+      const fQDeps = action.dependencies.map(act => {
+        const allActs = this.findActions(act);
+        return allActs.length === 1
+          ? allActs[0].proto.target.schema + "." + allActs[0].proto.target.name
+          : act;
```

Comment:
presumably we should throw a compileError if `allActs.length > 1`?

Reply:
This is probably captured elsewhere. The duplicity is checked in `checkTargetIsUnused` and in validation phase as a validation error. The tests pass, but I agree with you that this looks like the suitable place for throwing that error. I think there will be a lot more clarity as soon as I start moving the validation errors into compile.